cloudinit/cmd/status.py

```diff
@@ -69,54 +116,87 @@ def handle_status_args(name, args) -> int:
     """Handle calls to 'cloud-init status' as a subcommand."""
     # Read configured paths
     paths = read_cfg_paths()
-    status, status_detail, time = get_status_details(paths)
+    details = get_status_details(paths)
     if args.wait:
-        while status in (UXAppStatus.NOT_RUN, UXAppStatus.RUNNING):
-            sys.stdout.write(".")
-            sys.stdout.flush()
-            status, status_detail, time = get_status_details(paths)
+        while details.status in (UXAppStatus.NOT_RUN, UXAppStatus.RUNNING):
+            if args.format == "tabular":
+                sys.stdout.write(".")
+                sys.stdout.flush()
+            details = get_status_details(paths)
             sleep(0.25)
-        sys.stdout.write("\n")
-    print("status: {0}".format(status.value))
-    if args.long:
-        if time:
-            print("time: {0}".format(time))
-        print("detail:\n{0}".format(status_detail))
-    return 1 if status == UXAppStatus.ERROR else 0
+    details_dict: Dict[str, Union[str, List[str], Dict[str, Any]]] = {
+        "datasource": details.datasource,
+        "boot_status_code": details.boot_status_code.value,
+        "status": details.status.value,
+        "detail": details.description,
+        "errors": details.errors,
+        "last_update": details.last_update,
+    }
+    details_dict["schemas"] = {"1": copy.deepcopy(details_dict)}
+    details_dict["_schema_version"] = "1"
 
+    if args.format == "tabular":
+        prefix = "\n" if args.wait else ""
+        print(f"{prefix}status: {details.status.value}")
+        if args.long:
+            if details.last_update:
+                last_update = f"last_update: {details.last_update}\n"
+            else:
+                last_update = ""
+            print(
+                TABULAR_LONG_TMPL.format(
+                    prefix=prefix,
+                    boot_code=details.boot_status_code.value,
+                    description=details.description,
+                    last_update=last_update,
+                )
+            )
+    elif args.format == "json":
+        print(
+            json.dumps(  # Pretty, sorted json
+                details_dict, indent=1, sort_keys=True, separators=(",", ": ")
```

Comment:
Why 1? If we want it pretty, I think we should do at least 2 (if not 4).

Reply:
let's make it an arbitrary 2. easier for those humans to see, and avoids burning too many horizontal white space characters for small terminals.